docs/design/generics/terminology.md

```diff
@@ -637,60 +637,47 @@ conditions on the type argument. For example: `Array(T)` might implement
 
 ## Interface type parameters and associated types
 
-Imagine an interface defining a container. Different containers will contain
-different types of values, and the container API will have to refer to that
-"element type" when defining the signature of methods like "insert" or "find".
-If that element type is a parameter (input) to the interface type, we say it is
-an _interface type parameter_; if it is an output, we say it is an _associated
-type_. An associated type is a kind of [associated entity](#associated-entity).
-
-Interface type parameter example:
-
-```
-interface StackTP(ElementType:! Type)
-  fn Push[addr me: Self*](value: ElementType);
-  fn Pop[addr me: Self*]() -> ElementType;
-}
-```
-
-Associated type example:
+_Interface type parameters_ and _associated types_ are both ways of allowing the
+types in the signatures of function in an interface to vary. For example,
+different [stacks](<https://en.wikipedia.org/wiki/Stack_(abstract_data_type)>)
+will have different element types. That element type would be used as the
+parameter type of the `Push` function and the return type of the `Pop` function.
+An interface type parameter is a parameter or input to the interface type and
+associated types are outputs.
```

Comment:
I'm not sure I agree that AT are "outputs". They still have to be specified in order to implement an interface (though not when using it), which to me makes them inputs (or just integral part of the whole type). Did you mean "outputs" as-in "in the Stack example, it is the output type? Or maybe I'm missing something.

Reply:
I mean output as in an implementation determines values of the output types, as opposed to an interface parameter which needs to be specified to get an implementation. I'm trying to follow Rust's terminology: 
https://doc.rust-lang.org/rust-by-example/generics/assoc_items/types.html

Could you suggest some words that would make this clearer? Would it be useful to link to a Rust page about this?

Reply:
Another Rust link: https://rust-lang.github.io/rfcs/0195-associated-items.html#clearer-trait-matching

Reply:
Yes, I think it would help, especially the 2nd link. That's the PoV of someone who hasn't played with Rust yet, and is not used to rust terminology at least :).

Reply:
Updated: https://github.com/carbon-language/carbon-lang/pull/1972/commits/3b885158b17d5378f43804632a93d9544adb1dd2